app/src/main/java/space/taran/arknavigator/utils/FileUtils.kt

```diff
@@ -18,38 +33,132 @@ enum class Sorting {
     DEFAULT, NAME, SIZE, LAST_MODIFIED, TYPE
 }
 
-enum class FileActionType{
+enum class FileActionType {
     EDIT_AS_OPEN, EDIT_AND_OPEN, OPEN_ONLY, OPEN_ONLY_DETACH_PROCESS
 }
 
+enum class FileType {
+    IMAGE, VIDEO, GIF, PDF, UNDEFINED
+}
+
+const val PDF_PREVIEW_FOLDER_NAME = "pdf_preview"
+
 private val acceptedImageExt = listOf(".jpg", ".jpeg", ".png")
+private val acceptedVideoExt = listOf(".mp4", ".avi", ".mov", ".wmv", ".flv")
 private val acceptedEditOnlyExt = arrayListOf(".txt", ".doc", ".docx", ".odt", "ods")
     .also { it.addAll(acceptedImageExt) }
 private val acceptedReadAndEditExt = listOf(".pdf", ".md")
-private val acceptedVideoExt = listOf(".mp4", ".wmv", ".avi", ".flv", ".mkv")
-
-fun provideIconImage(file: Path): Path? =
-    providePreview(file) //todo downscale to, say, 128x128
-
-//might be a temporary file
-fun providePreview(file: Path): Path? =
-    if (isImage(file)) {
-        file
-    } else {
-        if (false) { //todo: create previews from videos, pdfs, etc.
-            TODO()
-        } else {
-            null
-        }
-    }
 
 fun isImage(filePath: Path): Boolean {
-    val extension = ".${extension(filePath)}"
+    val extension = extension(filePath)
     return acceptedImageExt.contains(extension)
 }
 
-fun getFileActionType(filePath: Path): FileActionType{
-    return when(".${extension(filePath)}"){
+fun isVideo(filePath: Path): Boolean {
+    if (!isValidFile(filePath)) return false
+
+    val extension = extension(filePath)
+    return acceptedVideoExt.contains(extension)
+}
+
+fun isValidFile(filePath: Path): Boolean {
+    return getFileSizeMB(filePath) != 0
+}
+
+fun isPDF(filePath: Path): Boolean {
+    return extension(filePath) == ".pdf"
+}
+
+fun isFormat(filePath: Path, format: String): Boolean {
+    return extension(filePath) == format
+}
+
+fun getPdfPreviewsFolder(): File =
+    File("${App.instance.cacheDir}/$PDF_PREVIEW_FOLDER_NAME")
+
+fun getPdfPreviewByID(id: Long): Path {
+    val pathName = "${App.instance.cacheDir}/$PDF_PREVIEW_FOLDER_NAME/$id.png"
+    return File(pathName).toPath()
+}
+
+fun getSavedPdfPreviews(): List<Long?>? =
+    getPdfPreviewsFolder()
+        .listFiles()
+        ?.map {
+            it.nameWithoutExtension.toLongOrNull()
+        }
+
+fun getVideoInfo(filePath: Path): MutableMap<Preview.ExtraInfoTag, String> {
```

Comment:
The same as with `createPdfPreview` — let's create separate file for video parsing.